flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/Schema.java

```diff
@@ -595,9 +621,57 @@ public Builder primaryKeyNamed(String constraintName, List<String> columnNames)
             return this;
         }
 
+        /**
+         * Declares an index for a set of given columns. Indexes are designed to enable very
+         * efficient search. The indexes are informational only and can be used for optimizations.
+         * It is the data owner's responsibility to guarantee the index queries allow the complete
+         * row to be retrieved efficiently.
+         *
+         * <p>The index will be assigned a generated name in the format {@code INDEX_col1_col2}.
+         *
+         * @param columns indexes that form a table index
+         */
+        public Builder index(String... columns) {
+            Preconditions.checkNotNull(indexes, "Index column names must not be null.");
+            return index(Arrays.asList(columns));
+        }
+
+        /**
+         * Declares an index for a set of given columns. Indexes are designed to enable very
+         * efficient search. The indexes are informational only and can be used for optimizations.
+         * It is the data owner's responsibility to guarantee the index queries allow the complete
+         * row to be retrieved efficiently.
+         *
+         * <p>The index will be assigned a generated name in the format {@code INDEX_col1_col2}.
+         *
+         * @param columns indexes that form a table index
+         */
+        public Builder index(List<String> columns) {
+            Preconditions.checkNotNull(indexes, "Index column names must not be null.");
+            final String generatedIndexName =
+                    columns.stream().collect(Collectors.joining("_", "INDEX_", ""));
+            return indexNamed(generatedIndexName, columns);
+        }
+
+        /**
+         * Declares an index for a set of given columns. Indexes are designed to enable very
```

Comment:
Declares an index for a set of given columns
->
Declares a named index for a list of given columns